src/main/kotlin/exercise5/BoardImpl.kt

```diff
@@ -1,13 +1,93 @@
 package exercise5
 
-// TODO Instantiate SquareBoard
-fun createSquareBoard(width: Int): SquareBoard = TODO("SquareBoardImpl(width)")
+fun createSquareBoard(width: Int): SquareBoard = SquareBoardImpl(width)
 
-// TODO Instantiate GameBoard
-fun <T> createGameBoard(width: Int): GameBoard = TODO("GameBoardImpl(width)")
+fun <T> createGameBoard(width: Int): GameBoard = GameBoardImpl(width)
 
-// TODO Implement SquareBoard Interface
-class SquareBoardImpl(size: Int)
+open class SquareBoardImpl(size: Int) : SquareBoard {
+    protected val board = List(size) { x ->
+        List(size) { y ->
+            Cell(x + 1, y + 1)
+        }
+    }
 
-// TODO extend SquareBoardImpl and implement GameBoard interface
-class GameBoardImpl(size: Int)
\ No newline at end of file
+    override val width = size
+
+    override fun getCellOrNull(i: Int, j: Int): Cell? {
+        if(i < 1 || i > width || j < 1 || j > width) return null
+
+        return board[i - 1][j - 1]
```

Comment:
Can be simplified as:
```suggestion
        return board.getOrNull(i - 1).getOrNull(j - 1)
```